code/controllers/subsystem/processing/station.dm

```diff
@@ -15,9 +15,10 @@ PROCESSING_SUBSYSTEM_DEF(station)
 	var/list/antag_protected_roles = list()
 	///A list of trait roles that should never be able to roll antag
 	var/list/antag_restricted_roles = list()
+	/// Assosciative list of goal type -> goal instance
+	var/list/datum/station_goal/goals_by_type = list()
```

Comment:
is it actually useful for this to be assoc? surely it's cleaner to just store em flat

Reply:
I prefer it :)